app/controllers/okta_controller.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+require 'openid_auth'
+
+class OktaController < ApplicationController
+  include OpenidAuth::V0
+  skip_before_action :authenticate
+
+  def metadata
+    meta = OneLogin::RubySaml::Metadata.new
+    render xml: meta.generate(saml_settings), content_type: 'application/xml'
+  end
+
+  def fetch_mvi_profile(json_request)
+    # TODO: pull out info from Okta request for querying MVI. Put into helper func
+    okta_attributes = json_request.dig(:data, :assertion, :claims)
+    user_attributes = Hash.new
+    okta_attributes.each do |k, v|
+      user_attributes[k] = v[:attributeValues][0][:value]
+    end
+    user_attributes["authn_context"] = json_request[:data][:assertion][:authentication][:authnContext][:authnContextClassRef]
```

Comment:
I don't think this is getting passed through, after closer examination. 

Validation error is coming from [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/mvi/service.rb#L133). We're sending the `mhv_icn`, so something is making the `.valid?` check return false